### PR TITLE
Migrate all ECC key operations to use `ethereum-keys` library.

### PR DESCRIPTION
**Pull request by pipermerriam**
### What was wrong?

New [`ethereum-keys`](https://github.com/pipermerriam/ethereum-keys) library does a better job of handling public and private keys and provides a simple api regardless of the backend.

### How was it fixed?

Switching over to that.  WIP

#### Cute Animal Picture

> put a cute animal picture here.

### CODE REVIEW

**Comment by pipermerriam** [comment]
I'm going to cleanup the commit history but I think this is ready for review.

One thing I don't like is the way I'm using the `KeyAPI` object.  It feels wrong which hints that it's the wrong API.  Very open to thoughts on how to improve it if you have any...

File: tests/p2p/test_kademlia.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
+import asyncio
```
Comment (gsalgado) [inline tests/p2p/test_kademlia.py]: Did you really intend to duplicate this test here? I'm keeping all the p2p stuff under evm/p2p to make it easier to eventually split it into its own repo.  Either way, you should at least remove this one and rename the original

File: evm/p2p/discovery.py [hunk 2]
```diff
@@ -250,9 +253,9 @@ def _unpack(message):
     message_hash = message[:MAC_SIZE]
     if message_hash != keccak(message[MAC_SIZE:]):
         raise WrongMAC()
-    signature = message[MAC_SIZE:HEAD_SIZE]
+    signature = KeyAPI().Signature(message[MAC_SIZE:HEAD_SIZE])
     signed_data = message[HEAD_SIZE:]
-    remote_pubkey = get_ecc_backend().ecdsa_recover(signed_data, signature)
+    remote_pubkey = signature.recover_msg(signed_data)
```
[the remark this answers is not in the document]
Comment (gsalgado) [reply evm/p2p/discovery.py]: Maybe `recover_pubkey_from_msg()`?

File: .travis.yml [hunk 3]
```diff
@@ -17,8 +17,8 @@ env:
     - TOX_POSARGS="-e py35-transactions"
     - TOX_POSARGS="-e py35-vm-fast"
     - TOX_POSARGS="-e py35-vm-limits"
-    - TOX_POSARGS="-e py35-coincurve"
     - TOX_POSARGS="-e py35-leveldb"
+    - TOX_POSARGS="-e py35-p2p"
```
[the remark this answers is not in the document]
Comment (pipermerriam) [reply .travis.yml]: Yes, I think so.

File: evm/p2p/discovery.py [hunk 4]
```diff
@@ -237,9 +240,9 @@ def _pack(cmd_id, payload, privkey):
     cmd_id = force_bytes(chr(cmd_id))
     expiration = rlp.sedes.big_endian_int.serialize(int(time.time() + EXPIRATION))
     encoded_data = cmd_id + rlp.encode(payload + [expiration])
-    signature = get_ecc_backend().ecdsa_sign(encoded_data, privkey)
-    message_hash = keccak(signature + encoded_data)
-    return message_hash + signature + encoded_data
+    signature = privkey.sign(encoded_data)
+    message_hash = keccak(bytes(signature) + encoded_data)
+    return message_hash + bytes(signature) + encoded_data
```
[the remark this answers is not in the document]
Comment (pipermerriam) [reply evm/p2p/discovery.py]: https://github.com/pipermerriam/ethereum-keys/pull/15

File: evm/p2p/discovery.py [hunk 5]
```diff
@@ -191,7 +194,7 @@ def send_neighbours(self, node, neighbours):
         nodes = []
         neighbours = sorted(neighbours)
         for n in neighbours:
-            l = n.address.to_endpoint() + [n.pubkey]
+            l = n.address.to_endpoint() + [bytes(n.pubkey)]
```
[the remark this answers is not in the document]
Comment (pipermerriam) [reply evm/p2p/discovery.py]: https://github.com/pipermerriam/ethereum-keys/pull/15

File: evm/p2p/discovery.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pipermerriam) [reply evm/p2p/discovery.py]: I like it.  I also think I'm going to change the `PrivateKey.sign` to be `PrivateKey.sign_msg` (to mirror the explicit `PrivateKey.sign_hash`.

**Pull request merged**